scripts/weekly-dep-report.sh

```diff
@@ -0,0 +1,90 @@
+    fi
+}
+
+get_total_deps () {
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind thirdparty --omit-edges-into diem-workspace-hack | wc -l | tr -d ' ')
+}
+
+get_total_direct_deps() {
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind directthirdparty --omit-edges-into diem-workspace-hack | wc -l | tr -d ' ')
+}
+
+get_total_dsr_deps () {
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind thirdparty --omit-edges-into diem-workspace-hack -p safety-rules | wc -l | tr -d ' ')
+}
+
+get_total_direct_dsr_deps () {
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind directthirdparty --omit-edges-into diem-workspace-hack -p safety-rules | wc -l | tr -d ' ')
+}
+
+get_total_dups () {
+    echo $(cargo +stable guppy dups --target x86_64-unknown-linux-gnu | wc -l | tr -d ' ')
+}
+
+
+git checkout --quiet "${last_commit}"
+date=$(git show --no-patch --format=%cd --date='format:%Y-%m-%d')
+short_sha=$(git rev-parse --short HEAD)
+total=$(get_total_deps)
+direct=$(get_total_direct_deps)
+lsr_total=$(get_total_dsr_deps)
+lsr_direct=$(get_total_direct_dsr_deps)
+dups=$(get_total_dups)
+
+if [[ -z "${first_commit_within_last_seven_days}" ]]; then
+    # case 1: no commit made in the last seven days
+    :
+else
+    # case 2: there is at least one commit in the last seven days
+    # assumption: there is at last one commit in the last week
+    last_commit_from_last_week=$(git rev-list "${first_commit_within_last_seven_days}"^1 -1)
+    git checkout --quiet "${last_commit_from_last_week}"
+fi
+
+prev_date=$(git show --no-patch --format=%cd --date='format:%Y-%m-%d')
+prev_short_sha=$(git rev-parse --short HEAD)
+prev_total=$(get_total_deps)
+prev_direct=$(get_total_direct_deps)
+prev_lsr_total=$(get_total_dsr_deps)
+prev_lsr_direct=$(get_total_direct_dsr_deps)
+prev_dups=$(get_total_dups)
+
+change_total=$(format_change_in_dependency_output $((total-prev_total)))
+change_direct=$(format_change_in_dependency_output $((direct-prev_direct)))
+change_lsr_total=$(format_change_in_dependency_output $((lsr_total-prev_lsr_total)))
+change_lsr_direct=$(format_change_in_dependency_output $((lsr_direct-prev_lsr_direct)))
+change_dups=$(format_change_in_dependency_output $((dups-prev_dups)))
+
+# resetting HEAD commit
+if [[ "${starter_point}" == "HEAD" ]]; then
+    git checkout --quiet "${last_commit}"
+else
+    git checkout --quiet "${starter_point}"
+fi
+
+echo "Dependency change in diem at Commit $short_sha ($date) since Commit $prev_short_sha ($prev_date) :"
```

Comment:
Specify more context, at least the branch, and the git slug.

Reply:
What is a git slug? just the repo name? Or do you mean the GitHub URL for this project?

Reply:
I'm not exactly sure what he means by git slug, but yeah it might be worthwhile to include the repo name or url so its clear this is from diem/diem